### PR TITLE
29 database restructuring for animation mode mapeditor

### PR DESCRIPTION
**Pull request by FJanssen-TNO**
Changed the results database structure to utilise the animation mode in the mapeditor.
I only need to check if the results of the conversion are acting as anticipated.

### CODE REVIEW

File: src/mesido/workflows/io/write_output.py [hunk 1]
```diff
@@ -904,6 +909,26 @@ def _name_to_asset(name):
                 try:
                     # If the asset has been placed
                     asset = _name_to_asset(asset_name)
+                    port, port_prim, port_sec = 3 * [None]
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:912]: 1. they are all initialised to be None, I just wanted them to be on one line.
2. yes, later I'm checking which got a value
3. allows me to set all the three variables to None. 3*[None] is the same as [None, None, None]

[the remark this answers is not in the document]
Comment (KobusVanRooyen) [reply src/mesido/workflows/io/write_output.py:912]: Sorry I think I am not clear. If you run in debug mode and put a breakpoint at line 913. You will see that all three variable are empty: port, port_prim, port_sec. They are not = [None, None, None] as expected. If this was the case then the append function would have been needed when adding/assigning values to these variables.

[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:912]: I indeed want all three variables to start empty, Here I'm just creating them all to be empty. They don't need to be list.

File: src/mesido/workflows/io/write_output.py [hunk 2]
```diff
@@ -892,6 +892,11 @@ def _name_to_asset(name):
             #     verify_ssl=self.influxdb_verify_ssl,
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:892]: updated

File: src/mesido/workflows/io/write_output.py [hunk 3]
```diff
@@ -892,6 +892,11 @@ def _name_to_asset(name):
...
             # )
 
+            profile_managers_carriers = {}
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:895]: My bad, forgot to remove it after changing the approach

File: src/mesido/workflows/io/write_output.py [hunk 4]
```diff
@@ -946,174 +971,225 @@ def _name_to_asset(name):
...
-                                    f"Asset {asset_name} has more than 1 OutPort, and the "
-                                    "profile data has been assigned to the 1st OutPort"
-                                )
-                                break
-
-                    if index_outport == -1:
-                        logger.error(
-                            f"Variable {index_outport} has not been assigned to the asset OutPort"
+                    if port:
+                        carrier_id_dict = {"carrier": port.carrier.id}
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:975]: Updated it together with the comment below.

File: src/mesido/workflows/io/write_output.py [hunk 5]
```diff
@@ -946,174 +971,225 @@ def _name_to_asset(name):
...
+                    elif port_prim and port_sec:
+                        carrier_id_dict = {
+                            "Primary": port_prim.carrier.id,
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:978]: updated

File: src/mesido/workflows/io/write_output.py [hunk 6]
```diff
@@ -946,174 +971,225 @@ def _name_to_asset(name):
...
-                                    measurement=asset_name,
-                                    field=profiles.profile_header[-1],
-                                    port=self.influxdb_port,
-                                    host=self.influxdb_host,
-                                    startDate=start_date_time,
-                                    endDate=end_date_time,
-                                    id=str(uuid.uuid4()),
+                        asset_class = asset.__class__.__name__
+                        asset_id = asset.id
+                        capabilities = [
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:1025]: updated

File: src/mesido/workflows/io/write_output.py [hunk 7]
```diff
@@ -946,174 +971,225 @@ def _name_to_asset(name):
...
+                            "Unsuported types for the different port carrier combinations"
                         )
-                        sys.exit(1)
 
-                    for ii in range(len(self.times())):
-                        if not self.io.datetimes[ii].tzinfo:
-                            data_row = [self.io.datetimes[ii].replace(tzinfo=datetime.timezone.utc)]
-                        else:
-                            data_row = [self.io.datetimes[ii]]
+                    variables_two_hydraulic_system_org = variables_two_hydraulic_system.copy()
```
[the remark this answers is not in the document]
Comment (FJanssen-TNO) [reply src/mesido/workflows/io/write_output.py:1011]: added

**Review by KobusVanRooyen — APPROVED** [review]

**Pull request merged**